### ISSUE TITLE
`lazygit-vscode.autoMaximizeWindow` closes sidebars when quitting

### ISSUE THREAD

**Issue by macintacos** [tom-pollak/lazygit-vscode#9]
Just discovered the `lazygit-vscode.autoMaximizeWindow` setting, which is great - however, it seems to close sidebars if they are open. Any way to disable this behavior? I'd prefer it if it just used the editor windows and left the positioning of sidebars alone.

**Comment by tom-pollak** [comment]
Unfortunately no.. I've tried to fix this for quite some time with no avail see [this stack overflow answer](https://stackoverflow.com/a/53926099/8973974)

> Also note that part of this v1.84 update is the REMOVAL of the older commands:
> ```
> View: Maximize Editor Group
> View: Unmaximize Editor Group
> ```

Luckily they didn't remove `workbench.action.maximizeEditorHideSidebar` (which is why your sidebar is hiding) and `workbench.action.evenEditorWidths` so I use these as a workaround.

If you want to have a crack at it code is [here](https://github.com/tom-pollak/lazygit-vscode/blob/1938c02f44dc4055a23a2701e4022b068ed89e4b/src/extension.ts#L190). Looking for either a `isMaximized(): boolean` function, a way of directly maximizing without hiding the sidebar, or a way of checking the state of the sidebar before so we can toggle it afterwards if necessary. The vscode api sucks 🤷

**Comment by tom-pollak** [comment]
Not sure what suspend is for actually, but lazygit.toggle should toggle the lazygit tab so keeps your current state until you close it

**Comment by jondkinney** [comment]
Claude AI says 

> Based on the available information, the suspend function in the lazygit editor_presets.go file appears to temporarily suspend the lazygit process and return control to the shell when an external editor is launched. This allows the user to edit files in their preferred editor while lazygit waits in the background, resuming once the editing is complete. 
The function is part of lazygit's editor integration system, which supports various editor presets as defined in the editor_presets.go file. 

Was just hoping there was an easier way to restore the file tree

**Comment by tom-pollak** [comment]
do you mean the "press enter to return to lazygit" popup? this isn't related to suspend but you can disable with:

```
promptToReturnFromSubprocess: false
```

**Comment by jondkinney** [comment]
No, I was talking about pressing `q` to exit LazyGit. It'd be nice if the Explorer view (sidebar with files) was re-opened. I was going to see if I could hack something together in keybindings.json with a context of "LazyGit". Not sure what the context would be, but was going to use inspect mode to try to find out. Then set `q` to a two actions: close lazygit, open explorer. Not sure if it'll work though.

**Comment by jondkinney** [comment]
I took a look at the context in the dev console and didn't see any mention of LazyGit at all. There's plenty about Gitlens though! Hah! Maybe you could add a boolean to context like `lazygit.isOpen` and set that to true or false, then we could add a keybinding for `q` that would reference that context and override `q` only when lazygit is open? Then run whatever commands we want to restore windows or the explorer, chat panel, etc. We'd need a way to pass the regular `q` through though first, then have a short timeout 🤔 Just throwing out some ideas.

**Comment by tom-pollak** [comment]
ah I see what you mean. I tried to fix this ran into a wall with VSCodes [bad public api](https://github.com/tom-pollak/lazygit-vscode/issues/9#issuecomment-2555751863) a fix would be appreciated! what's feasible is a config setting that opens the side panel after `workbench.action.evenEditorWidths`, although this would open the sidebar if it wasn't open before toggling lazygit.

**Comment by jondkinney** [comment]
I took a shot at it here https://github.com/tom-pollak/lazygit-vscode/pull/15

It's working the way I want, at least. LMK what you think!

**Comment by tom-pollak** [comment]
done in
